File: 1678-goal-parser-interpretation/1678-goal-parser-interpretation.swift
class Solution {
    func interpret(_ command: String) -> String {
        var result = ""
        var i = 0
        
        while i < command.count {
            let c = command[command.index(command.startIndex, offsetBy: i)]
            
            if c == "G" {
                result += String(c)
                i += 1
                continue
            }
            if c == "(" {
                let d = command[command.index(command.startIndex, offsetBy: i + 1)]
                if d == ")" {
                    result += "o"
                    i += 2
                } else {
                    result += "al"
                    i += 4
                }
            }
        }
        
        
        return result
    }
}

File: 1678-goal-parser-interpretation/1678-goal-parser-interpretation.rs
impl Solution {
    pub fn interpret(command: String) -> String {
        let mut result = String::new();

        let mut c = command.chars();

        while let Some(cal) = c.next() {
            if cal == 'G' {
                result.push(cal);
                continue;
            }
            let n = c.next();

            if let Some(next_val) = n {
                if next_val == ')' && cal != 'l' {
                    result.push('o');
                }
                if next_val == 'a' {
                    result.push_str("al");
                }
            }
        }

        result
    }
}
